pkg/dyninst/decode/marshal.go

```diff
@@ -352,13 +352,7 @@ func (d *Decoder) collectSwissMapTables(
 	// Deduplicate addrs by sorting and then removing duplicates.
 	// Go swiss maps may have multiple table pointers for the same group.
 	slices.Sort(addrs)
-	dedupedAddrs := addrs[:0]
-	for i, addr := range addrs {
-		if i == 0 || addr != addrs[i-1] {
-			dedupedAddrs = append(dedupedAddrs, addr)
-		}
-	}
-	addrs = dedupedAddrs
+	addrs = slices.Compact(addrs)
 	tableType, ok := s.TablePtrSliceType.Element.(*ir.PointerType)
```

Comment:
you could and perhaps should precompute this assertion and all the other type assertions. I get how that leads use back to the problem of cycle decoderType references, but I think in practice it's trivial to side-step.